cmake/helper_functions.cmake

```diff
@@ -12,6 +12,14 @@ function(CREATE_EAGER_TEST_EXE TESTLIST)
             set(TEST_TARGET ${TEST_NAME})
         endif()
         add_executable(${TEST_TARGET} ${TEST_SRC_PATH})
+        set_target_properties(
+            ${TEST_TARGET}
+            PROPERTIES
+                UNITY_BUILD
+                    ON
+                UNITY_BUILD_UNIQUE_ID
```

Comment:
This was introduced in CMake v3.20.  Officially we're still supporting [v3.16](https://github.com/tenstorrent/tt-metal/blob/main/CMakeLists.txt#L1).  I'm not certain whether unrecognized target properties are silently ignored or error out.  I intend to add a static check to catch any backwards-incompatible additions... Any Day Now 😅 

But also, these all have the same value.  Why not just set CMAKE_UNITY_BUILD_UNIQUE_ID in our root and have done?  That would be backwards compatible without question.

Reply:
Thanks for noticing! I bumped the minimum required version to 3.20. This feature is required for a successful build. Also I moved out this code into a function and reused it across codebase

Reply:
Are we at liberty to bump the minimum?  If so, let's go with v3.22, which is the version that ships with Ubuntu 22.04.

Right now I've been cautious about bumping the required version because officially we're only supporting Ubuntu 20.04, and requiring a newer CMake is friction for customers.  We are actively working on getting to 22.04 (and then a fast follow to 24.04).  But the roadmap is still lengthy to get to 22.04.

I have bumped CMake to v3.22 on CI in the 20.04 image for some optional features, so _we_ are fine with newer.  But I don't know where our customers are right now.